modules/graph/fragment/arrow_fragment.grin.h

```diff
@@ -0,0 +1,978 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef MODULES_GRAPH_FRAGMENT_ARROW_FRAGMENT_MOD_H_
+#define MODULES_GRAPH_FRAGMENT_ARROW_FRAGMENT_MOD_H_
+
+#include <cstddef>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "arrow/api.h"
+#include "arrow/io/api.h"
+
+#include "grape/fragment/fragment_base.h"
+#include "grape/graph/adj_list.h"
+#include "grape/utils/vertex_array.h"
+
+#include "client/ds/core_types.h"
+#include "client/ds/object_meta.h"
+
+#include "basic/ds/arrow.h"
+#include "basic/ds/arrow_utils.h"
+#include "common/util/typename.h"
+
+#include "graph/fragment/arrow_fragment_base.h"
+#include "graph/fragment/fragment_traits.h"
+#include "graph/fragment/graph_schema.h"
+#include "graph/fragment/property_graph_types.h"
+#include "graph/utils/error.h"
+#include "graph/vertex_map/arrow_local_vertex_map.h"
+#include "graph/vertex_map/arrow_vertex_map.h"
+
+extern "C" {
+#include "grin/include/topology/structure.h"
+#include "grin/include/topology/vertexlist.h"
+#include "grin/include/topology/edgelist.h"
+#include "grin/include/topology/adjacentlist.h"
+#include "grin/include/partition/partition.h"
+#include "grin/include/propertygraph/label.h"
+#include "grin/include/propertygraph/property.h"
+#include "grin/include/propertygraph/propertygraph.h"
+}
+
+namespace gs {
+
+template <typename OID_T, typename VID_T, typename VDATA_T, typename EDATA_T,
+          typename VERTEX_MAP_T>
+class ArrowProjectedFragment;
+
+}  // namespace gs
+
+namespace vineyard {
+
+inline std::string generate_name_with_suffix(
+    const std::string& prefix, property_graph_types::LABEL_ID_TYPE label) {
+  return prefix + "_" + std::to_string(label);
+}
+
+inline std::string generate_name_with_suffix(
+    const std::string& prefix, property_graph_types::LABEL_ID_TYPE v_label,
+    property_graph_types::LABEL_ID_TYPE e_label) {
+  return prefix + "_" + std::to_string(v_label) + "_" + std::to_string(e_label);
+}
+
+template <typename OID_T, typename VID_T, typename VERTEX_MAP_T>
+class ArrowFragmentBaseBuilder;
+
+template <typename OID_T, typename VID_T,
+          typename VERTEX_MAP_T =
+              ArrowVertexMap<typename InternalType<OID_T>::type, VID_T>>
+class __attribute__((annotate("vineyard"))) ArrowFragment
+    : public ArrowFragmentBase,
+      public vineyard::BareRegistered<
+          ArrowFragment<OID_T, VID_T, VERTEX_MAP_T>> {
```

Comment:
If I understand it correctly, grin should be an access interface and we don't need some `vineyard::Object`s there?

I was imaging some thing like

```
std::shared_ptr<arrow::Fragment> fragment = client.GetObject<vineyard::ArrowFragment>("objectid");

auto g /* grin::Fragment */ = grin::g(fragment);
```